core/src/main/java/hudson/model/Executor.java

```diff
@@ -843,7 +845,13 @@ public HttpResponse doStop() {
         lock.writeLock().lock(); // need write lock as interrupt will change the field
         try {
             if (executable != null) {
-                Tasks.getOwnerTaskOf(getParentOf(executable)).checkAbortPermission();
+                final SubTask parentOf;
+                try {
+                   parentOf = getParentOfOrFail(executable);
+                } catch(InvocationTargetException ex) {
+                    return HttpResponses.error(500, ex);
```

Comment:
`Executor.doStop` is actually used from plugins AFAIK, so should not be restricted. Better to leave its original implementation alone unless this is somehow required to fix the reported bug.